sympy/matrices/common.py

```diff
@@ -203,6 +203,12 @@ def col_insert(self, pos, other):
         if not self:
             return type(self)(other)
 
+        try:
```

Comment:
Is there a reason to prefer that TypeError instead of ValueError be raised? If not, this could just be `pos = as_int(pos)` and the standard ValueError would be raised...and it should give enough of a clue as to what is wrong.

Reply:
@smichr the purpose of using the `TypeError` was to clarify to the user what exactly has he done wrong and what is the correct format. `ValueError` only shows `pos is not an integer` while `TypeError` also shows what is the correct format.

Reply:
@smichr  What do you think?

Reply:
I think we should not create custom error messages if a standard response is sufficient. e.g.
```
>>> factorint(3.5)
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/base/data/home/apps/s~sympy-live-hrd/55.412087474737415211/sympy/sympy/ntheory/factor_.py", line 1029, in factorint
    n = as_int(n)
  File "/base/data/home/apps/s~sympy-live-hrd/55.412087474737415211/sympy/sympy/core/compatibility.py", line 377, in as_int
    raise ValueError('%s is not an integer' % (n,))
ValueError: 3.5 is not an integer
```

The user learns from this that 3.5 is not an integer. They also know that this is happening in factorint which they just called. So saying that "factorint expects an integer" is redundant. My feeling is that the same applies here. Any routine that expects to process integers and needs to raise an error for non-integer intput can safely put `n = as_int(n)` at the top of their routine and know that no non-int will pass and the user will know of the issue.